### ISSUE TITLE
kubectl get rs <name> -w fails with error BadRequest (unknown reason)

### ISSUE THREAD

**Issue by girishkalele** [kubernetes/kubernetes#45568]
**Error from server (BadRequest): the server rejected our request for an unknown reason (get replicasets.extensions)**

I don't know how to get logs from the apiserver running on GKE master.

```
$ kubectl get rs nginx1-3103140165 -w
NAME                DESIRED   CURRENT   READY     AGE
nginx1-3103140165   10        10        10        2h
Error from server (BadRequest): the server rejected our request for an unknown reason (get replicasets.extensions)
```

**Kubernetes version** (use `kubectl version`):

```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"6", GitVersion:"v1.6.2", GitCommit:"477efc3cbe6a7effca06bd1452fa356e2201e1ee", GitTreeState:"clean", BuildDate:"2017-04-19T20:33:11Z", GoVersion:"go1.7.5", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"6", GitVersion:"v1.6.2", GitCommit:"477efc3cbe6a7effca06bd1452fa356e2201e1ee", GitTreeState:"clean", BuildDate:"2017-04-19T20:22:08Z", GoVersion:"go1.7.5", Compiler:"gc", Platform:"linux/amd64"}
```

**Environment**:

GKE v1.6.2

**What happened**:

- This works `$ kubectl get rs <name>`
- This does not work `$ kubectl get rs <name> -w`

But it works without an rs specified.

- This works `$ kubectl get rs -w`

**What you expected to happen**:

We expect that kubectl should watch replicaset named <name> for changes.

**Comment by xingzhou** [comment]
cc @caesarxuchao and @miminar
The cause of this issue is because API server does not have field label conversion function for `ReplicaSet`. 
According to this line, https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/extensions/v1beta1/conversion.go#L51
I can only see the conversion functions for `DaemonSet`, `Deployment` and `Ingress` are set. As a result, when kubectl or client request the following url
```
curl -k -v -XGET  -H "Accept: application/json" -H "User-Agent: kubectl/v1.4.0 (linux/amd64) kubernetes/575f254" http://localhost:8080/apis/extensions/v1beta1/namespaces/default/replicasets?fieldSelector%3Dnginx-1253638599
```
API server will respond with BadRequest error. If you think we can supplement `ReplicaSet` to the `extensions/v1beta1` list, I'd like to submit a PR then.

**Comment by mml** [comment]
ping @caesarxuchao

**Comment by liggitt** [comment]
fixed by #51197
